tests/unit/commands/test_remote.py

```diff
@@ -523,6 +526,101 @@ def test_run_in_repo_newer_than_core22(
     emitter.assert_debug("Running new remote-build because base is newer than core22")
 
 
+@pytest.mark.parametrize(
+    "create_snapcraft_yaml", LEGACY_BASES | {"core22"}, indirect=True
+)
+@pytest.mark.usefixtures("create_snapcraft_yaml", "mock_confirm", "mock_argv")
+def test_run_in_shallow_repo(emitter, mock_run_legacy, new_dir):
+    """core22 and older bases run new remote-build if in a git repo."""
+    root_path = Path(new_dir)
+    git_normal_path = root_path / "normal"
+    git_normal_path.mkdir()
+    git_shallow_path = root_path / "shallow"
+
+    shutil.move(root_path / "snap", git_normal_path)
+
+    repo_normal = GitRepo(git_normal_path)
+    (repo_normal.path / "1").write_text("1")
+    repo_normal.add_all()
+    repo_normal.commit("1")
+
+    (repo_normal.path / "2").write_text("2")
+    repo_normal.add_all()
+    repo_normal.commit("2")
+
+    (repo_normal.path / "3").write_text("3")
+    repo_normal.add_all()
+    repo_normal.commit("3")
+
+    # pygit2 does not support shallow cloning, so we use git directly
+    subprocess.run(
+        [
+            "git",
+            "clone",
+            "--depth",
+            "1",
+            git_normal_path.absolute().as_uri(),
+            git_shallow_path.absolute().as_posix(),
+        ],
+        check=True,
+    )
+
+    os.chdir(git_shallow_path)
+    cli.run()
+
+    mock_run_legacy.assert_called_once()
+    emitter.assert_debug("Current git repository is shallow cloned.")
+    emitter.assert_progress("Fallback to legacy remote-build", permanent=True)
+
+
+@pytest.mark.parametrize("create_snapcraft_yaml", {"devel"}, indirect=True)
```

Comment:
Why only `devel`?

Reply:
You will get invalid base if you put `core24`, which it may need update in another PR

Reply:
We can do this and add `core24` to `CURRENT_BASES` in another PR.
```suggestion
@pytest.mark.parametrize("create_snapcraft_yaml",  CURRENT_BASES - {"core22"}, indirect=True)
```

Reply:
That's more future proof